client/internal/dns/server.go

```diff
@@ -45,6 +45,7 @@ type DefaultServer struct {
 	runtimePort        int
 	runtimeIP          string
 	previousConfigHash uint64
+	currentConfig      nbdns.Config
```

Comment:
Keeping the whole DNS config might backfire in smaller devices as we might have `CustomZones` with thousands of records. So it might be better to only keep nbdns.Config.NameServerGroups as it will be in most cases, small.

Reply:
Fixed,
Using [hostDNSConfig](https://github.com/netbirdio/netbird/pull/676/commits/8856e3c2963f773d4fe9db13ba8b724d3c0f01f8) to save state instead of `nbdns.Config`.